### PR TITLE
Add descriptions on doc.go for common/compiler package

### PR DESCRIPTION
**Pull request by andybclee**
## Proposed changes

- Added more desciption on doc.go of the common/compiler package
- This is a documentation related PR

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ehnuje — DISMISSED** [review]
LGTM

**Review by jimni1222 — DISMISSED** [review]
LGTM!

File: common/compiler/doc.go [hunk 1]
```diff
@@ -18,5 +18,9 @@
 // This file is derived from common/compiler/solidity.go (2018/06/04).
 // Modified and improved for the klaytn development.
 
-// Package compiler wraps the Solidity compiler executable (solc).
+/*
+Package compiler wraps the Solidity compiler executable (solc).
+
+This package is being used by `abigen` and test codes and provides functions to compile Solidity source codes
```
Comment (jimni1222) [inline common/compiler/doc.go]:
```suggestion
This package is being used by `abigen` and test codes and provides functions to compile Solidity source codes.
```

Comment (laggu) [inline common/compiler/doc.go]: Would be better use `This package is used`?

Comment (andybclee) [reply common/compiler/doc.go]: I see. Actually repetitive use of 'and' doesn't look good.

**Review by ehnuje — APPROVED** [review]
LGTM

**Review by jimni1222 — APPROVED** [review]
LGTM !

**Comment by kaiachain-dev** [comment]
<details><summary>Builds ready [a8b7a99]</summary><ul><li>Linux: <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kcn-v1.0.2~rc.1-0-linux-amd64.tar.gz>kcn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kpn-v1.0.2~rc.1-0-linux-amd64.tar.gz>kpn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ken-v1.0.2~rc.1-0-linux-amd64.tar.gz>ken</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kscn-v1.0.2~rc.1-0-linux-amd64.tar.gz>kscn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kspn-v1.0.2~rc.1-0-linux-amd64.tar.gz>kspn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ksen-v1.0.2~rc.1-0-linux-amd64.tar.gz>ksen</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kbn-v1.0.2~rc.1-0-linux-amd64.tar.gz>kbn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kgen-v1.0.2~rc.1-0-linux-amd64.tar.gz>kgen</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/homi-v1.0.2~rc.1-0-linux-amd64.tar.gz>homi</a> </li><li>Darwin: <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kcn-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kcn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kpn-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kpn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ken-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>ken</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kscn-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kscn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kspn-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kspn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ksen-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>ksen</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kbn-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kbn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kgen-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kgen</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/homi-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>homi</a> </li><li>Baobab-linux: <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kcn-baobab-v1.0.2~rc.1-0-linux-amd64.tar.gz>kcn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kpn-baobab-v1.0.2~rc.1-0-linux-amd64.tar.gz>kpn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ken-baobab-v1.0.2~rc.1-0-linux-amd64.tar.gz>ken</a> </li><li>Baobab-darwin: <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kcn-baobab-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kcn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/kpn-baobab-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>kpn</a> <a href=http://packages.klaytn.net/klaytn/v1.0.2~rc.1/ken-baobab-v1.0.2~rc.1-0-darwin-10.10-amd64.tar.gz>ken</a> </li></ul></details>

**Pull request merged**